interpreter/Makefile

```diff
@@ -142,78 +55,47 @@ TESTDIR =	../test/core
 TESTFILES =	$(shell cd $(TESTDIR); ls *.wast; ls [a-z]*/*.wast)
 TESTS =		$(TESTFILES:%.wast=%)
 
-.PHONY:		test debugtest partest dune-test
+.PHONY:		test partest dune-test
 
 test:		$(OPT) smallint
-		$(TESTDIR)/run.py --wasm `pwd`/$(OPT) $(if $(JS),--js '$(JS)',)
-		./smallint.native
-debugtest:	$(UNOPT) smallint
-		$(TESTDIR)/run.py --wasm `pwd`/$(UNOPT) $(if $(JS),--js '$(JS)',)
-		./smallint.native
+		$(TESTDIR)/run.py --wasm `pwd`/$(BUILDDIR)/$(OPT) $(if $(JS),--js '$(JS)',)
+		dune exec ./smallint.exe
 
 test/%:		$(OPT)
-		$(TESTDIR)/run.py --wasm `pwd`/$(OPT) $(if $(JS),--js '$(JS)',) $(TESTDIR)/$*.wast
-debugtest/%:	$(UNOPT)
-		$(TESTDIR)/run.py --wasm `pwd`/$(UNOPT) $(if $(JS),--js '$(JS)',) $(TESTDIR)/$*.wast
+		$(TESTDIR)/run.py --wasm `pwd`/$(BUILDDIR)/$(OPT) $(if $(JS),--js '$(JS)',) $(TESTDIR)/$*.wast
 
 run/%:		$(OPT)
 		./$(OPT) $(TESTDIR)/$*.wast
-debug/%:	$(UNOPT)
-		./$(UNOPT) $(TESTDIR)/$*.wast
 
-partest: 	$(TESTS:%=quiettest/%)
+partest:	$(TESTS:%=quiettest/%)
 		@echo All tests passed.
 
 quiettest/%:	$(OPT)
 		@ ( \
-		  $(TESTDIR)/run.py 2>$(@F).out --wasm `pwd`/$(OPT) $(if $(JS),--js '$(JS)',) $(TESTDIR)/$*.wast && \
+		  $(TESTDIR)/run.py 2>$(@F).out --wasm `pwd`/$(BUILDDIR)/$(OPT) $(if $(JS),--js '$(JS)',) $(TESTDIR)/$*.wast && \
 		  rm $(@F).out \
 		) || \
 		cat $(@F).out || rm $(@F).out || exit 1
 
 smallinttest:	smallint
-		@./smallint.native
+	dune exec ./smallint.exe
 
 dunetest:
 	dune test
 
+install:
+	dune build -p $(NAME) @install
+	dune install
 
 # Miscellaneous targets
 
 .PHONY:		clean
 
-$(ZIP):		$(WINMAKE)
-		git archive --format=zip --prefix=$(NAME)/ -o $@ HEAD
+$(ZIP):
+	git archive --format=zip --prefix=$(NAME)/ -o $@ HEAD
 
 clean:
-		rm -rf _build/jslib $(LIB).mlpack _tags $(JSLIB).js
-		$(OCB) -clean
-
-
-# Opam support
-
-.PHONY:		check install uninstall
-
-check:
-		# Check that we can find all relevant libraries
-		# when using ocamlfind
-		ocamlfind query $(LIBS)
-
-install:	_build/$(LIB).cmx _build/$(LIB).cmo
-		ocamlfind install $(LIB) meta/findlib/META _build/$(LIB).o \
-		  $(wildcard _build/$(LIB).cm*) \
-		  $(wildcard $(DIRS:%=%/*.mli))
-
-uninstall:
-		ocamlfind remove $(LIB)
+	dune clean
 
-opam-release/%:
-		git tag opam-$*
-		git push --tags
-		rm -f opam-$*.zip
-		wget https://github.com/WebAssembly/spec/archive/opam-$*.zip
-		cp meta/opam/opam .
-		sed -i ".tmp" s/@VERSION/$*/g opam
-		sed -i ".tmp" s/@MD5/`md5 -q opam-$*.zip`/g opam
-		rm opam.tmp opam-$*.zip
-		@echo Created file ./opam, submit to github opam-repository/packages/wasm/wasm.$*/opam
```

Comment:
Thanks for volunteering, I am happy to pass that on to you. :)

Yet, others should still be able to do that easily in case, to avoid single points of failure. A makefile target is a useful abstraction that is both automating and documenting the steps necessary, and tends to be a more stable interface than the tools behind it. (Once a person has worked on a dozen diverse systems that each have gone through multiple generations of custom build tools, they start to appreciate having that abstraction. ;) )

If opam-publish or dune-release are the best way of implementing that process now (I admit I haven't used those before), can you adjust the Makefile target to call into those?